program/tests/deactivate.rs

```diff
@@ -0,0 +1,128 @@
+#![allow(clippy::arithmetic_side_effects)]
+
+mod helpers;
+
+use {
+    helpers::{context::StakeTestContext, lifecycle::StakeLifecycle},
+    mollusk_svm::result::Check,
+    solana_account::ReadableAccount,
+    solana_program_error::ProgramError,
+    solana_pubkey::Pubkey,
+    solana_stake_client::instructions::{DeactivateBuilder, DelegateStakeBuilder},
+    solana_stake_interface::{error::StakeError, state::StakeStateV2},
+    solana_stake_program::id,
+    test_case::test_case,
+};
+
+#[test_case(false; "activating")]
+#[test_case(true; "active")]
+fn test_deactivate(activate: bool) {
```

Comment:
Not for this PR, but when we later migrate this instruction, I think it would help to revisit these tests and likely break them up into specific cases versus a single test with multiple inside.